lib/ansible/plugins/callback/yaml.py

```diff
@@ -113,7 +113,7 @@ def _dump_results(self, result, indent=None, sort_keys=True, keep_invocation=Fal
 
         if abridged_result:
             dumped += '\n'
-            dumped += yaml.dump(abridged_result, width=1000, Dumper=AnsibleDumper, default_flow_style=False)
+            dumped += yaml.dump(abridged_result, allow_unicode=True, width=1000, Dumper=AnsibleDumper, default_flow_style=False).decode('utf-8')
```

Comment:
This looks like it would throw an exception on Python3 (because yaml.dump will return a text string and you can't decode a text string on Python3).

Since what we want to do is make sure that it is a text string on both Python2 and Python3, what we should do is use to_text here to transform it into a text string:
``` python
from ansible.module_utils._text import to_text
[....]
dumped += yaml.dump(to_text(abridged_result, allow_unicode=True, width=1000, Dumper=AnsibleDumper, default_flow_style=False))
```

Reply:
I missed where you're doing the .decode.... the new version of the line should look like this:
``` python
dumped += to_text(yaml.dump(abridged_result, allow_unicode=True, width=1000, Dumper=AnsibleDumper, default_flow_style=False))
```

Reply:
Thanks for that hint. Switched to `to_text`..